clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -5506,16 +5509,13 @@ Sema::GetMissingFormatAttributes(Stmt *Body, const FunctionDecl *FDecl) {
     }
 
     // Get first argument index
-    unsigned FirstToCheck = [&]() -> unsigned {
+    int FirstToCheck = [&]() -> unsigned {
       if (!FDecl->isVariadic())
         return 0;
-      const auto *FirstToCheckArg =
-          dyn_cast<DeclRefExpr>(Args[NumArgs - 1]->IgnoreParenCasts());
```

Comment:
I removed dynamic cast to DeclRefAttr of last function argument. In C++ mode casting to DeclRefAttr resulted in unrecognizing argument type as va_list for several architectures and OS.